rfcs/rfc-story-based-git-workflow.md

```diff
@@ -0,0 +1,117 @@
+- Start Date: 2019-09-24
+- Target Major Version: N/A
+- Reference Issues: N/A
+- Entity Issue: 
+- Implementation PR: 
+
+# Summary
+
+This RFC proposes an approach to code management in the master branch of [https://github.com/bcgov/lear](https://github.com/bcgov/lear), specifically around branching, code reviews, and merging pull requests into the master branch.
+
+The proposal is to use a story branch workflow to increase the stability of the master branch by merging larger, more complete, and fully tested batches of work. The team agreement proposed is to shoot for a commit history on master that reflects changes story-by-story, where each story is complete and has been tested. 
+
+# Basic example
```

Comment:
Can this be shorter?
Reference our dev workflow that is documented under namex?
1. Dev team create a branch for their collective story.
2. Complete development tasks, and meet delivery standards (linting, tests, code cov, etc.)
3. Rebase as required, squash commits to logical units and submit PR

Reply:
It's not about whether we branch or merge, it's about merging larger, more complete, and fully tested batches of work. Agree or disagree (which is the point of an RFC in the first place). 

No disagreement with the dev workflow. I think branches off master have some advantages to branches off shared forks, but that's not the point of the RFC.

Reply:
Actually, the namex workflow document does not address the core issue which is collaborative work outside of master. It's very much "one developer at a time".